.github/actions/javascript/reviewerChecklist/reviewerChecklist.js

```diff
@@ -0,0 +1,81 @@
+const core = require('@actions/core');
+const github = require('@actions/github');
+const https = require('https');
+const GitHubUtils = require('../../../libs/GithubUtils');
+
+const pathToReviewerChecklist = 'https://raw.githubusercontent.com/Expensify/App/main/contributingGuides/REVIEWER_CHECKLIST.md';
+const reviewerChecklistStartsWith = '## Reviewer Checklist';
+const issue = github.context.payload.issue ? github.context.payload.issue.number : github.context.payload.pull_request.number;
+const combinedComments = [];
+
+/**
+ * @returns {Promise}
+ */
+function getNumberOfItemsFromReviewerChecklist() {
+    return new Promise((resolve, reject) => {
+        https.get(pathToReviewerChecklist, (res) => {
+            let fileContents = '';
+            res.on('data', (chunk) => {
+                fileContents += chunk;
+            });
+            res.on('end', () => {
+                const numberOfChecklistItems = (fileContents.match(/- \[ \]/g) || []).length;
+                resolve(numberOfChecklistItems);
+            });
+        })
+            .on('error', reject);
+    });
+}
+
+/**
+ * @param {Number} numberOfChecklistItems
+ */
+function checkIssueForCompletedChecklist(numberOfChecklistItems) {
+    GitHubUtils.getAllReviewComments(issue)
+        .then(reviewComments => combinedComments.push(...reviewComments))
+        .then(() => GitHubUtils.getAllComments(issue))
+        .then(comments => combinedComments.push(...comments))
+        .then(() => {
+            let foundReviewerChecklist = false;
+            let numberOfFinishedChecklistItems = 0;
+            let numberOfUnfinishedChecklistItems = 0;
+
+            // Once we've gathered all the data, loop through each comment and look to see if it contains the reviewer checklist
+            for (let i = 0; i < combinedComments.length; i++) {
+                // Skip all other comments if we already found the reviewer checklist
+                if (foundReviewerChecklist) {
```

Comment:
Is there ever a situation where someone would re-comment with a new version of the checklist, rather than updating the earlier one? Probably not, right?

Reply:
I don't think I've seen a case of that, no. That would be a pretty weird flow for someone to do.

Reply:
Yeah agreed. Just double checking!